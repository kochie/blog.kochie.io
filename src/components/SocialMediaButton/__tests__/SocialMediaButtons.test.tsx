import { IconName, IconPrefix } from '@fortawesome/fontawesome-svg-core'
import React from 'react'
import { describe, test, expect } from 'vitest'

import SMButton from '..'
import { render } from '@testing-library/react'

describe('Social Media Button Component', () => {
  test('renders correctly', () => {
    const props = {
      name: 'Twitter',
      link: 'https://twitter.com',
      icon: ['fab', 'twitter'] as [IconPrefix, IconName],
      color: '#1DA1F2',
      tracking: 'TEST',
    }

    const { asFragment } = render(<SMButton sm={props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
