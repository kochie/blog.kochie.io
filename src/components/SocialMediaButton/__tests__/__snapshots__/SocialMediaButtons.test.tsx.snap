// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Social Media Button Component > renders correctly 1`] = `
<DocumentFragment>
  <div>
    <a
      class="text-white transition ease-in-out duration-200"
      href="https://twitter.com"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-twitter fa-1x mx-1 transform-gpu transition duration-200 ease-in-out"
        data-icon="twitter"
        data-prefix="fab"
        role="img"
        viewBox="0 0 512 512"
      >
        <path
          d="M459.4 151.7c.3 4.5 .3 9.1 .3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103l0-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53 51.7 63.7 129.3 105.3 216.4 109.8-1.6-7.8-2.6-15.9-2.6-24 0-57.8 46.8-104.9 104.9-104.9 30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
